src/bin/pants/main.rs

```diff
@@ -11,97 +11,62 @@
-        .short("s")
-        .long("pants_style")
-        .takes_value(true)
-        .help("Your pants style"))
-      .arg(Arg::with_name("loud")
-        .short("d")
-        .long("loud")
-        .takes_value(false)
-        .help("Also show non-vulnerable dependencies"))
-      .arg(Arg::with_name("no-color")
-        .short("m")
-        .long("no-color")
-        .takes_value(false)
-        .help("Disable color output"))
-      .arg(common::get_verbose_arg().clone())
-      .arg(common::get_lockfile_arg().clone())
-      .arg(common::get_dev_arg().clone())
-    )
-    .get_matches();
-
-    match matches.subcommand() {
-        ("pants", Some(sub_m)) => {
-            let log_level = common::get_log_level_filter(sub_m);
+mod cli;
 
+fn main() {
+    let opt = cli::Opt::from_args();
+
+    match opt {
+        cli::Opt::Pants {
+            toml_file,
+            log_level,
+            include_dev_dependencies,
+            loud,
+            no_color,
+            pants_style,
+            oss_index_api_key,
+        } => {
             common::construct_logger(false, log_level);
 
-            handle_pants_sub_command(sub_m);
-        }
-        _ => common::print_no_command_found("pants".to_string()),
-    }
-}
-
-fn handle_pants_sub_command(pants_matches: &ArgMatches) {
-    if let Some(pants_style) = pants_matches.value_of("pants_style") {
-        check_pants(pants_style);
-    }
-
-    let toml_file_path = pants_matches.value_of("tomlfile").unwrap();
-    let verbose_output = pants_matches.is_present("loud");
-    let enable_color: bool = !pants_matches.is_present("no-color");
-    let dev: bool = pants_matches.is_present("dev");
-
-    common::print_dev_dependencies_info(dev);
-
-    audit(
-        toml_file_path.to_string(),
-        verbose_output,
-        enable_color,
-        dev,
-    );
-}
+            if let Some(pants_style) = pants_style {
+                check_pants(&pants_style);
+            }
 
-fn get_api_key() -> Option<String> {
-    match env::var("OSS_INDEX_API_KEY") {
-        Ok(val) => return Some(val),
-        Err(e) => {
-            info!("Warning: missing optional 'OSS_INDEX_API_KEY': {}", e);
+            common::print_dev_dependencies_info(include_dev_dependencies);
 
-            return None;
+            audit(
+                toml_file.to_string_lossy().to_string(),
+                oss_index_api_key,
+                loud,
+                !no_color,
+                include_dev_dependencies,
+            );
         }
-    };
+    }
 }
 
-fn audit(toml_file_path: String, verbose_output: bool, enable_color: bool, include_dev: bool) -> ! {
+fn audit(
+    toml_file_path: String,
+    oss_index_api_key: Option<String>,
+    verbose_output: bool,
+    enable_color: bool,
+    include_dev: bool,
+) -> ! {
     let mut parser = ParseCargoToml::new(toml_file_path.clone(), include_dev);
     let packages = match parser.get_packages() {
         Ok(packages) => packages,
```

Comment:
also please ignore similar to prior `Err(e)` question:

IDEA is flagging this (line 74 - not shown in diff), but it I'm not sure it makes sense to me.

     `Error` doesn't implement `Display` (required by {})

Just asking for my own knowledge - not a blocker seems to me. Dunno if we exercise this code.